src/app/client/src/app/modules/observation/components/observation-details/observation-details.component.ts

```diff
@@ -101,37 +106,152 @@ export class ObservationDetailsComponent implements OnInit {
     this.getEntities();
   }
   goBack() {
-    this.router.navigate(["/observation"]);
+    this.router.navigate(['/observation']);
   }
 
-  observeAgain() {
-    const paramOptions = {
-      url:
-        this.config.urlConFig.URLS.OBSERVATION.OBSERVATION_SUBMISSION_CREATE +
-        `${this.observationId}?entityId=${this.selectedEntity._id}`,
-      param: {},
-      data: {
-        block: "0abd4d28-a9da-4739-8132-79e0804cd73e",
-        district: "2f76dcf5-e43b-4f71-a3f2-c8f19e1fce03",
-        role: "DEO",
-        school: "8be7ecb5-4e35-4230-8746-8b2694276343",
-        state: "bc75cc99-9205-463e-a722-5326857838f8",
-      },
-    };
-    this.observationService.post(paramOptions).subscribe((data) => {
-      console.log(data, "data 122");
-      this.getObservationForm();
+  async observeAgain() {
+    let metaData = await this.observationUtilService.getAlertMetaData();
+    metaData.content.body.data = this.resourceService.frmelmnts.lbl.createObserveAgain;
+    metaData.content.body.type = "text";
+    metaData.content.title = this.resourceService.frmelmnts.btn.observeAgain;
+    metaData.size = "mini";
+    metaData.footer.buttons.push({
+      type: "cancel",
+      returnValue: false,
+      buttonText: this.resourceService.frmelmnts.btn.no
     });
+    metaData.footer.buttons.push({
+      type: "accept",
+      returnValue: true,
+      buttonText: this.resourceService.frmelmnts.btn.yes
+    })
+    metaData.footer.className = "double-btn-circle";
+    let returnData = await this.observationUtilService.showPopupAlert(metaData);
+    console.log(returnData, "returnData")
+    if (returnData) {
+      this.showLoader = true;
+      const paramOptions = {
+        url: this.config.urlConFig.URLS.OBSERVATION.OBSERVATION_SUBMISSION_CREATE + `${this.observationId}?entityId=${this.selectedEntity._id}`,
+        param: {},
+        data: this.payload,
+      };
+      this.observationService.post(paramOptions).subscribe(data => {
+        console.log(data, "data 122");
+        this.showLoader = false;
+        this.getObservationForm();
+      }, error => {
+        this.showLoader = false;
+      })
+    }
   }
-
-  redirectToQuestions(submissionNumber,evidenceCode) {    
+  openObservation(event) {
     this.router.navigate([`/questionnaire`], {
       queryParams: {
         observationId: this.observationId,
         entityId: this.selectedEntity._id,
-        submissionNumber: submissionNumber,
-        evidenceCode: evidenceCode,
-      },
+        submissionNumber: event.code
```

Comment:
event.code should be assigned to the evidenceCode key not to the submisisonNumber